lib/improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -342,37 +335,35 @@ def pressure_correct_svp(svp, temperature, pressure):
             Series, Vol. 30; Equation A4.7.
 
         Args:
-            svp (iris.cube.Cube):
+            svp (numpy.ndarray):
                 A cube of saturated vapour pressures (Pa).
-            temperature (iris.cube.Cube):
+            temp (numpy.ndarray):
                 A cube of air temperatures (K, converted to Celsius).
-            pressure (iris.cube.Cube):
+            pressure (numpy.ndarray):
                 Cube of pressure (Pa).
 
         Returns:
-            svp (iris.cube.Cube):
+            svp (numpy.ndarray):
                 The input cube of saturated vapour pressure of air (Pa) is
                 modified by the pressure correction.
         """
-        temp = temperature.copy()
-        temp.convert_units('celsius')
```

Comment:
No check to convert data to Celsius.

Reply:
That is taken to outside the function.
on line 374 (at time of writing)
```python
        temp = temp.data + consts.ABSOLUTE_ZERO
        svp = self.pressure_correct_svp(svp, temp, pressure)
```